### DISCUSSION TITLE
c# override class name where x-parser-schema-id is present

### DISCUSSION THREAD

**Discussion by 3p05** [asyncapi/modelina#1510] [Q&A] [answered]
I'm generating my first c# models from an AsyncAPI document.

The AsyncAPI document has many occurrences of the x-parser-schema-id property, all set to anonymous-schema-<x>.

The models generated are using this the value of x-parser-schema-id, which will be confusing to use and we would prefer to use the name property.

How can we configure this?

Thanks

**Comment by jonaslagoni** [comment]
To more accurately define names for models in your AsyncAPI document, you can set the `$id` or `title` property. Or alternatively, you can set propper `x-parser-schema-id` values so they reflect the right names 🙂 

You can also overwrite the naming constrainer: https://modelina.org/examples?selectedExample=overwrite-naming-formatting

**Reply by 3p05** [reply]
Thanks for getting back to me.

The AsyncAPI document I'm trying to generate the models from isn't mine, so I cant make any changes to it, which is less than ideal. This is also my first time in dealing with AsyncAPI or modelina.

Most of the elements do already have the `name` and `title` properties set, but it seems that `x-parser-schema-id` is being used over name. This means there's over 230 anonymous schema elements.

I can see that in the example you sent, the model name is passed and then modified, but I can't see how or where to either pass the model, or to specify name over x-parser-schema-id. 

It's also possible I'm not using modelina correctly. as a noob to AsyncAPI and Modelina the documentation was not the simplest to get started with. I've taken one of the examples and pasted in the AsyncAPI document. I've attached the index file in case you can suggest something I'm doing wrong. 

[index-ts.txt](https://github.com/asyncapi/modelina/files/12716691/index-ts.txt)

Thanks again for your help.

**Reply by jonaslagoni** [reply]
Hmm, yea I see your problem. None of the suggestions I made solves your problem..

Let me think about that for a sec. I'll get back to you tomorrow 🤔

**Reply by jonaslagoni** [reply, marked as answer]
@3p05 I think there are two ways forward here.

1st one is that you want `message.name` to be used as the root payload model name, right? If that is the case please create a feature request 🙂 

2nd you want to have further control over `what` value is selected as the model name in the first place, i.e. in this case you don't want `x-parser-schema-id` but something else. I would create another feature request for this as well 🙂 

Furthermore, the AsyncAPI document you have access to really is not optimal for code generation. Cause we have to use best effort to try and guess the names for the generated models, because one was not provided in the schema. Whoever is the owner of the document needs to change it. Or you create your own with your own namings of schemas. The tools can only do so much.

It very much follows garbage in, garbage out principle 😅
